semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -401,8 +401,24 @@ let m_with_symbolic_propagation ~is_root f b =
       && not is_root)
     ~then_:
       (match b.G.e with
-      | G.N (G.Id (_, { id_svalue = { contents = Some (G.Sym b1) }; _ })) ->
-          f b1
+      | G.N (G.Id ((id, _), { id_svalue = { contents = Some (G.Sym b1) }; _ }))
+        ->
+          (* We shouldn't end up with a symbol that resolves to itself, but if
```

Comment:
This doesn't seem like a proper fix, may be fine to merge for now but we need a Linear ticket to keep track of the issue and make sure it's properly fixed.

Reply:
I guess that it's either a bug in name resolution, which is giving both occurrences of `foo` the same `id_info` when it should not, or it is a bug in symbolic-propagation. Since these bugs have pretty bad consequences, perhaps we should add a guardrail here and bound the amount of `id_svalue`s that we can follow. I would prefer that we add that sanity check now in this PR but I understand you may be pressed for time. However please create the Linear ticket and assign it to me.

Reply:
The test case I have for DeepSemgrep is `const foo = foo` (see my PR there) which is invalid code to begin with. I'm not really worried about analyzing it correctly (whatever that means) -- just worried about not crashing. I'll put together a linear ticket for you, though -- I think the original test case was more reasonable.

I agree this isn't a complete solution, but I think there should be guardrails here anyway. I'd be fine if you'd like to limit depth instead, but as you've noted I am a bit pressed for time at the moment.